### ISSUE TITLE
My Keyboard change to an other Layout

### ISSUE THREAD

**Issue by LeChatNoir666** [LizardByte/Sunshine#493]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Is your issue described in the documentation?

- [X] I have read the documentation

### Describe the Bug

If I connect to my Desktop-PC my Keyboard changed from German to somewhat US-like, so Z becomes Y.
I searched the options of the Moonlight client and in the Web-UI of Sunshine but didn't found an option for this.
The Windows keyboard layout didn't changed so I wonder about this behavior.

### Expected Behavior

Letting my keyboard layout untouched.

### Host Operating System

Windows

### Operating System Version

10

### Architecture

64 bit

### Sunshine commit or version

0.15.0

### Package

Windows - portable

### GPU Model

Radeon 580

### GPU Driver/Mesa Version

22.5.1

**Comment by ld892012** [comment]
You can do this until this bug is fixed :
[https://github.com/loki-47-6F-64/sunshine/issues/82](url)

**Comment by gschintgen** [comment]
Hi, I'm on Sunshine 0.18 and my keyboard input (swiss french layout on client & host) is still messed up in a very strange way: Some keys (shift + digit) are fine, whereas others are mixed up (e.g. 'é' is to the right of 'L', but via moonlight&sunshine it arrives on the host as 'è') or broken in the classical qwertz vs qwerty way where a 'z' becomes a 'y'. Entering passwords is fun ;-)

Should I file a new issue?

The behavior that I described above is the one I'm observing right now on sunshine 0.18 windows with moonlight 4.3.1 windows. (I could also check with moonlight-embedded on RPi4. If it's deemed useful I may also set up sunshine/moonlight on my main non-gaming desktop, which is running Ubuntu.)

**Comment by gschintgen** [comment]
Hmm, nevermind. Just now I was in the process of properly reproducing it (1st checking that it's fine on GFE, installing sunshine nightly, etc.) only to find that it's now working properly. It must have been some kind of fluke that probably could have been resolved by a reboot.

**Comment by FlorianWolters** [comment]
For reference: https://github.com/LizardByte/Sunshine/discussions/989#discussioncomment-8355575

**Comment by FlorianWolters** [comment]
Did anyone find a fully working solution for a German QWERTZ keyboard to use correct key mappings with Sunshine, Windows 11 and Moonlight Android Client (NVIDIA Shield TV Pro 2019)?

I have tried deactivation of "send scancodes" (which breaks ingame experience completely). I have also tried to use keybindings, but was not able yet to create a fully working mapping. While the following does fix some keys, a lot keys (normal press and in combination with ALT GR and SHIFT) are still incorrect.

This is the only mayor annoyance I do currently have with Sunshine, otherwise this software is awesome.

Here are my incomplete keybindings trying to fix German QWERTZ keyboard:
    keybindings = [
        0xbd, 0xdb,
        0xbb, 0xdd,
        0x59, 0x5a,
        0xdb, 0xba,
        0xdd, 0xbb,
        0xba, 0xc0,
        0xc0, 0xdc,
        0xdc, 0xbf,
        0x5a, 0x59,
        0xbf, 0xbd
    ]

**Comment by FlorianWolters** [comment]
I have just noticed that Sunshine does **not** use a QWERTY US keyboard on my system (that's what I thought until now), but does something pretty weird. It only messes up some particular keys. So far I noticed the following, which is incorrect on my German QWERTZ keyboard:

- `Y` and `Z` are swapped.
- `?ß\` and `-_` are swapped.
- `<>|` is mapped to `´`` (i.e. the later is placed on two keys)

This is pretty weird, but at least I do have a hint what I've to try to replace via the `keybindings` settings. This looks like a bug to me, should I report a new issue?
